rasa/utils/tensorflow/model_data.py

```diff
@@ -1039,6 +1039,18 @@ def _create_label_ids(label_ids: FeatureArray) -> np.ndarray:
 
         raise ValueError("Unsupported label_ids dimensions")
 
+    @staticmethod
+    def _filter_4d_arrays(array_of_array_of_features):
+        return list(
+            filter(
+                lambda x: len(x) > 0,
+                [
+                    list(filter(lambda x: x.shape[0] > 0, array_of_features))
```

Comment:
can't we use the parameter `number_of_dimension`?

Reply:
what do you mean? `number_of_dimension` is always `4` here, it's just sequence dim might be `0`

Reply:
ah, misread the function name, can you please add a comment? thanks